### ISSUE TITLE
Bun seems to be always invoking `node` on a MacBook Pro M2

### ISSUE THREAD

**Issue by diraneyya** [oven-sh/bun#13903]
### What version of Bun is running?

1.1.27+267afa293

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

Using `asdf` for tooling version management, any of the following create commands will trigger an `asdf` warning asking for a version of `node` to be activated within the project:
```bash
bun create react-app ./bun-react-app
bun create vite ./bun-vite-app
bun create hono ./bun-hono-app
```

### What is the expected behavior?

The expected behaviour is for `bun` to be a replacement of both `node` and `npm`.

The promise:
<img width="767" alt="image" src="https://github.com/user-attachments/assets/6f5c2c36-fc92-46a4-aec2-c7bbb4f6750b">

### What do you see instead?

<img width="1157" alt="image" src="https://github.com/user-attachments/assets/c75bf6bd-8a5e-4919-9616-b9938ec5e365">
<img width="1162" alt="image" src="https://github.com/user-attachments/assets/c99858d3-a723-4d1c-bd5f-c0ff90c538a9">
<img width="1174" alt="image" src="https://github.com/user-attachments/assets/4c101370-cf61-46ef-9b7a-24e7f2249d01">

### Additional information

I have tried to run many repositories from GitHub using bun, none of them worked, with similar failures. For example, running `bun install` on the [food-nutrients](https://github.com/food-nutrients/food-nutrients) bun app from GitHub results in an error message `error: postinstall script from "esbuild" exited with 126`. The commands `bun run dev` and `bun run build` also seem to fail while asking me to setup node within the project.

`neofetch` output:
> OS: macOS 13.4.1 22F82 arm64 
> Host: Mac14,9 
> Kernel: 22.5.0 
> Uptime: 16 days, 40 mins 
> Packages: 263 (brew) 
> Shell: zsh 5.9 
> Resolution: 1512x982 
> DE: Aqua 
> WM: Quartz Compositor 
> WM Theme: Blue (Light) 
> Terminal: Apple_Terminal 
> Terminal Font: FiraCodeNerdFontCompleteM-Retina 
> CPU: Apple M2 Pro 
> GPU: Apple M2 Pro 
> Memory: 3203MiB / 16384MiB

**Comment by pfgithub** [comment]
use "--bun" to symlink "bun" to "node" while running scripts, or run without node installed and it will always use bun

**Comment by chrisbbreuer** [comment]
<img width="413" alt="Screenshot 2024-09-12 at 11 06 52 AM" src="https://github.com/user-attachments/assets/639d6601-8b44-4001-91ba-323c1331ce56">
or use this in your bunfig

**Comment by diraneyya** [comment]
Thank you both for this. I remember reading about the `--bun` option in the documentation, I just seem to have confused the default state. What I recalled is that calling bun was the default, and that we were able to allow scripts to call `node` if that was the intention.

Now that I am learning that it is the other way around, I am able to install and run things without `node`, finally.

I noticed that the `[run.bun]` config does not work reliably, and that it only applies to `bun run`.

As for `bun install` and `bun create`, using `--bun` seems to work for now.

Thank you! I will close this issue.
